src/gluonts/model/san/_engine.py

```diff
@@ -0,0 +1,302 @@
+from typing import Optional, List, Callable, Iterator
+import os
+import logging
+import tempfile
+import inspect
+import uuid
+import time
+
+import numpy as np
+import mxnet as mx
+from mxnet.gluon import nn, HybridBlock, Trainer as MXTrainer
+
+from gluonts.mx.trainer import Trainer as BaseTrainer
+from gluonts.mx.trainer.learning_rate_scheduler import MetricAttentiveScheduler
+from gluonts.mx.trainer.model_averaging import (
+    AveragingStrategy,
+    SelectNBestMean,
+    save_epoch_info,
+)
+from gluonts.model.forecast import QuantileForecast
+from gluonts.model.forecast_generator import (
+    QuantileForecastGenerator as BaseForecastGenerator,
+)
+from gluonts.dataset.loader import (
+    DataLoader,
+    TrainDataLoader,
+    ValidationDataLoader,
+    InferenceDataLoader,
+)
+from gluonts.dataset.field_names import FieldName
+from gluonts.dataset.common import DataEntry
+from gluonts.gluonts_tqdm import tqdm
+from gluonts.support.util import HybridContext
+
+logger = logging.getLogger("gluonts").getChild("trainer")
+MODEL_ARTIFACT_FILE_NAME = "model"
+STATE_ARTIFACT_FILE_NAME = "state"
+LOG_CACHE = set([])
+OutputTransform = Callable[[DataEntry, np.ndarray], np.ndarray]
+
+
+class Trainer(BaseTrainer):
```

Comment:
Do you need to overload the `Trainer` class here?

Reply:
Because the base trainer do not accept default argument values in hybrid_forward. I overload the trainer to give default None to unused feature types

Reply:
Let's try to update to use the `Trainer` class or extend it within there.